src/towncrier/build.py

```diff
@@ -106,6 +106,16 @@ def _validate_answer(ctx: Context, param: Option, value: bool) -> bool:
     help="Do not ask for confirmations. But keep news fragments.",
     callback=_validate_answer,
 )
+@click.option(
```

Comment:
I prefer to have this CLI option inside `towncrier check`

I think that for `towncrier build` this should be the default.

That is, `towncrier build` should always fail when unknown fragments are found.

--------

and maybe we don't need a `towncrier check` fragment.

Just make this the defaul and add an option to explicly ignore selected files inside the `newsfragments` directory.

Reply:
Sounds good

Reply:
Note that my suggestion about the default behaviour for `towncrier build` might not be backward compatible .

I was just talking about the ideal case.

------

So we might need to add a configuration like this `build_ignore_filenames = ["README.rst", "template.j2"]`

and then fail by default, only if `build_ignore_filenames` is not empty.

and `build_ignore_filenames` should be empty by default

This should give us backward compatibility

-------

and then we can have `towncrier build --strict` that will fail if there are extra files, even when `build_ignore_filenames` is empty.

Reply:
So `towncrier build` should have a `--strict` flag, but should it also have an opposite flag so that even if `build_ignore_filenames` is set, a user can ignore any invalid filenames?

Also I'm a bit confused about what you would like for `towncrier check`, is it the same flags as for `towncrier build`?

Reply:
I think that `towncrier check` don't need any flag. It should just fail by default for invalid fragments.

The backward compatibility should be for `towncrier build`.

I think that for `towncrier check` people will expect that newer version of `towncrier` will introduce more checks ... so you might end up with a check that fails.

Reply:
Hmm, I'm not sure this works, because if you haven't set `config.build_ignore_filenames` then `towncrier check` will fail on the template file.

Reply:
> Hmm, I'm not sure this works, because if you haven't set config.build_ignore_filenames then towncrier check will fail on the template file.

The failure should be ok.
If `towncrier check` fails, you will need to udpate `config.build_ignore_filenames`
The error message might include a hint that the file should be added to `towncrier` ignore liss

-----

The template file might be stored outside of the `newsfragments` directory,
in which case, all is ok.

But most probably we will alwasy have a ".keep" or ".gitignore" file that should be ignored by default

For example this is what we have for `towncrier` and similar for `twisted`

https://github.com/twisted/towncrier/blob/trunk/src/towncrier/newsfragments/.gitignore

Reply:
Maybe we can remove this CLI argument and just use the configuration to control the behaviour of `build`